### ISSUE TITLE
2015-06-30 Remove wind trend from NLDAS

### ISSUE THREAD

**Issue by lawinslow** [USGS-R/mda.lakes#70]
## How does RMSE and bias change when wind trend is removed from NLDAS data?

**Comment by lawinslow** [comment]
## What do the data look like? 

The full timeseries of NLDAS wind data aggregated across a bunch (100) of our lake sites.

![image](https://cloud.githubusercontent.com/assets/347600/8464232/5a1246fe-2006-11e5-9be4-7f707b0baa5e.png)

It looks to be a step change, not a trend. The average annual values across our 1979-2012 range look like this
![image](https://cloud.githubusercontent.com/assets/347600/8464286/9aef6ada-2006-11e5-9a79-4a3d457ac523.png)

The change occurs between 2001 and 2002. Hard to visually see a difference in distribution
![image](https://cloud.githubusercontent.com/assets/347600/8464322/cc47d4e6-2006-11e5-994b-e97115603bc8.png)

But we can see it when we fit the data to a weibull distribution. 
```R
> fitdist(subset(sp, year>2001)$WindSpeed, 'weibull')
Fitting of the distribution ' weibull ' by maximum likelihood 
Parameters:
      estimate Std. Error
shape 3.005518 0.03401985
scale 5.830320 0.03101341

> fitdist(subset(sp, year<2001)$WindSpeed, 'weibull')
Fitting of the distribution ' weibull ' by maximum likelihood 
Parameters:
      estimate Std. Error
shape 3.012133  0.0247867
scale 5.372733  0.0210710
````
You can see the difference in the scale factor. 

## How do we handle this
I propose we scale the wind data after 2001 using a multiplication factor. (Proposed value, 0.921, which is the quotient of the before and after 2001 scale factors). 

I'm going to see how RMSE changes if we do this. Trip report later.

**Comment by lawinslow** [comment]
As point of interest, here is the weibull paramters for scaled wind speed from NTL-LTER airport met station. 

```R
library(fitdistrplus)
library(LakeMetabolizer)
> fitdist(head(wind.scale.base(wnd$wnd, 3), 30000), 'weibull')
Fitting of the distribution ' weibull ' by maximum likelihood 
Parameters:
      estimate  Std. Error
shape 1.636043 0.007673722
scale 3.690860 0.013661206
````
Lower than we get with NLDAS

**Comment by jread-usgs** [comment]
wow. I wonder if the NLDAS folks know about that step change. I will see if there is a forum or something.

**Comment by lawinslow** [comment]
Full cal run with NLDAS. Through 2012 (note, comparisons with downscaled climate models will differ as they are only compared through 1999 because of model time coverage)

Sim | RMSE | Bias
---- | ---- | ----
Standard NLDAS | 2.966868 | 0.351693
Step Fix NLDAS | XX | XX

**Comment by jread-usgs** [comment]
If you are only scaling after 2001 and this is only pre-1999, will the numbers differ?

**Comment by lawinslow** [comment]
No. I'm running this through 2012.

**Comment by jread-usgs** [comment]
That initial rmse is from the full sim? Thought it was 2.6 or 2.73?

**Comment by lawinslow** [comment]
Ignore that. Those numbers aren't final.

**Comment by lawinslow** [comment]
Ok, the increase in bias suggests that perhaps the winds after 2001 are actually more characteristic than those before 2001 (though decrease RMSE suggests otherwise??). 

Let's see what happens if we flip this correction and adjust the winds before 2001.

```R
  bfr_2001 = nldas$time < as.POSIXct('2001-12-31')
  nldas$WindSpeed[bfr_2001] = nldas$WindSpeed[bfr_2001] * 1.0857
````

**Comment by jread-usgs** [comment]
wild. You running the inverse now?

**Comment by lawinslow** [comment]
Yup

**Comment by lawinslow** [comment]
Sim | RMSE | Bias
---- | ---- | ----
Standard NLDAS |  2.584166 | 0.06114929
Step Fix NLDAS | 2.553134 | 0.1457446
Pre-2001 step fix | 2.612 | -0.00242

> mean(all_cal$Observed_wTemp - all_cal$Modeled_wTemp, na.rm=TRUE)
[1] -0.002428753
> 
> sqrt(mean((all_cal$Observed_wTemp - all_cal$Modeled_wTemp)^2, na.rm=TRUE))
[1] 2.612053

**Comment by jread-usgs** [comment]
smaller bias, larger error. moving target

**Comment by jread-usgs** [comment]
@lawinslow just curious - when you do bias, do you get a different picture if you only look at the best 95% of the residuals? Just wondering if extreme outliers have much of an impact on the bias numbers.

**Comment by lawinslow** [comment]
Not really. For that last one, if you drop the outer 10%, RMSE goes to `2.37043`
```R
resids = na.omit(all_cal$Observed_wTemp - all_cal$Modeled_wTemp)
med = median(resids)
high = med + 2*(quantile(resids, 19/20, na.rm=TRUE) - med) 
low = med + 2* (quantile(resids, 1/20, na.rm=TRUE) - med)
inner_resids = resids[resids< high & resids > low] 
sqrt(mean(inner_resids^2, na.rm=TRUE))

````

**Comment by jread-usgs** [comment]
and bias isn't affected?

**Comment by lawinslow** [comment]
A little, but not much. `-0.01318243` What's your thought process on this?

**Comment by jread-usgs** [comment]
I guess I was thinking that similar to how we dealt with the modeling errors before, we should toss out the *really* bad fits, so they don't steer us off. But, it doesn't look like there is enough of a change to make this something that should be applied. 

That said, I assume the distribution fitting is pretty robust, but does it result in similar mean(wind^3) for the season? Since the power is the cube, those tails are really important, which is probably what you were seeing in the plot above with mean(wind) for the season. Do you feel confident that the shift is appropriate for the data?

**Comment by lawinslow** [comment]
Good question. I'll take a look.

**Comment by lawinslow** [comment]
Using this correction  
```R
after_2001 = nldas$time > as.POSIXct('2001-12-31')
nldas$WindSpeed[after_2001] = nldas$WindSpeed[after_2001] * 0.921
```
and then calculating annual averages of wind work (wind^3). Red is raw NLDAS. Blue is fixed. 

![image](https://cloud.githubusercontent.com/assets/347600/8585775/5da793fe-25ac-11e5-9b0b-44de4c337e6f.png)

**Comment by jread-usgs** [comment]
nice! curious how the sims turn out w/ this.

**Comment by lawinslow** [comment]
### ME Trends Before wind correction
![nldas grid sens me](https://cloud.githubusercontent.com/assets/347600/8600927/62a8d80e-262e-11e5-9973-fa9fadea06cf.png)

### ME Trends after correction
![nldas windfix grid sens me](https://cloud.githubusercontent.com/assets/347600/8600943/736410a0-262e-11e5-8571-5d04210c405f.png)

### SP Trends before wind correction
![nldas depths sens sp](https://cloud.githubusercontent.com/assets/347600/8600945/7c6ab848-262e-11e5-89fe-13252ee2ab4b.png)

### SP Trends after correction
![nldas windfix grid sens sp](https://cloud.githubusercontent.com/assets/347600/8600955/890283f6-262e-11e5-82d4-303f18d60da8.png)

### SP Empirical Trends (since 1984)
![grid yday empir sens sp](https://cloud.githubusercontent.com/assets/347600/8601005/d8afeea2-262e-11e5-8899-6b8c85c9d76a.png)

**Comment by jread-usgs** [comment]
@lawinslow my question earlier was about the bias - I thought the table above was from a different correction, which didn't do the scaling according to average wind work. did you have the bias and rmse numbers for that?

**Comment by lawinslow** [comment]
@jread-usgs How do you do the scaling according to average wind work?

**Comment by jread-usgs** [comment]
this: @lawinslow 
![image](https://cloud.githubusercontent.com/assets/2349007/8621050/12c74454-26e9-11e5-899f-bb58940eab27.png)

**Comment by lawinslow** [comment]
Ok, I thought you meant there was some way to scale based on wind^3 (like `(0.921*wind^3)^1/3` or something). Do you see the RMSE and bias values [here](https://github.com/USGS-R/mda.lakes/issues/70#issuecomment-119244878)?

**Comment by jread-usgs** [comment]
yes I see those, but I am asking if those apply to the 0.921 correction or the other corrections that seem to distribution based.

**Comment by jread-usgs** [comment]
I will look at this w/ NARR for WI

**Comment by lawinslow** [comment]
There's only one style of bias correction I've used (the multiplicative one, applied to either pre-2001 or post-2001 data). The distribution work was used to objectively calculate the 0.921 coefficient.

**Comment by jread-usgs** [comment]
ok, makes sense now.
